File: Engine/src/main/java/org/etieskrill/engine/graphics/texture/animation/TextureAnimationMetadata.java
package org.etieskrill.engine.graphics.texture.animation;

import lombok.Data;
import org.etieskrill.engine.graphics.texture.AbstractTexture;
import org.joml.Vector2ic;

import java.util.List;

@Data
public class TextureAnimationMetadata {
    private final String textureFile;
    private final Vector2ic frameSize;
    private final AbstractTexture.Format format;
    private final List<TextureAnimationFrame> frames;
    private final float duration;
}

File: Engine/src/commonMain/kotlin/org/etieskrill/engine/graphics/texture/animation/TextureAnimationMetadata.kt
package org.etieskrill.engine.graphics.texture.animation

import org.etieskrill.engine.graphics.texture.TextureFormat
import org.joml.Vector2ic

data class TextureAnimationMetadata(
    val textureFile: String,
    val frameSize: Vector2ic,
    val format: TextureFormat,
    val frames: List<TextureAnimationFrame>,
    val duration: Float
)
